features/step_definitions/book_steps.rb

```diff
@@ -0,0 +1,67 @@
+# require "pry-byebug"
+Before do
+  visit("/users/sign_up")
+  fill_in "Email", :with => "test@example.com"
+  fill_in "Password", :with => "password"
+  fill_in "Password confirmation", :with => "password"
+  click_button("Sign up")
+  open_email("test@example.com")
+  visit_in_email("Confirm my account")
+  @book = build(:book)
+end
+
+Given(/^there are (\d+) books in the database$/) do |amount|
+  amount.to_i.times { create(:book) }
+end
+
+Given(/^I am logged into the site$/) do
+  visit("/users/sign_in")
+  fill_in "Email", :with => "test@example.com"
+  fill_in "Password", :with => "password"
+  click_button("Log in")
+end
+
+When(/^I visit the root url$/) do
+  create(:book, published_date: "2015-09-27")
+  visit("/")
+end
+
+Then(/^I see a list of books in the database$/) do
+  expect(page).to have_content(Book.first.title)
+end
+
+Then(/^the books are ordered by published date$/) do  
+  expect(page).to have_content("2015-09-27")
```

Comment:
This is good.

See if there's a way to inspect the `@books` instance variable and see that the one you want is `@books.first`.